cmd2/utils.py

```diff
@@ -520,10 +520,15 @@ def send_sigint(self) -> None:
         """Send a SIGINT to the process similar to if <Ctrl>+C were pressed."""
         import signal
         if sys.platform.startswith('win'):
-            signal_to_send = signal.CTRL_C_EVENT
+            self._proc.send_signal(signal.CTRL_C_EVENT)
         else:
-            signal_to_send = signal.SIGINT
-        self._proc.send_signal(signal_to_send)
+            # Since cmd2 uses shell=True in its Popen calls, we need to send the SIGINT to
+            # the whole process group to make sure it propagates further than the shell
+            try:
+                group_id = os.getpgid(self._proc.pid)
```

Comment:
Is `self._proc` and `self._proc.pid` guaranteed to exist?  Would it be better as the following?
```Python
group_id = os.getpgid(os.getpid())
```

I think the current process pid is guaranteed to exist, so you might not need to wrap it in the `try / except ProcessLookupError` if you do it the above way?

Reply:
`ProcReader._proc` is a `subprocess.Popen` object and is passed into `ProcReader` at initialization. This code is attempting to send a SIGINT to its process group, and not the current process' group. The `ProcessLookupError` is there because `os.getpgid()` and `os.killpg()` can fail if the subprocess has already ended.